folium/features.py

```diff
@@ -1992,3 +2002,54 @@ def __init__(
             out.setdefault(cm(color), []).append([[lat1, lng1], [lat2, lng2]])
         for key, val in out.items():
             self.add_child(PolyLine(val, color=key, weight=weight, opacity=opacity))
+
+
+class Control(JSCSSMixin, MacroElement):
+    """
+    Add a Leaflet Control object to the map
+
+    Parameters
+    ----------
+    control: str
+        The javascript class name of the control to be rendered.
+    position: str
+        One of "bottomright", "bottomleft", "topright", "topleft"
```

Comment:
One problem that we have in folium templates are the silent erros when passing keywords to it. For example, if someone types `bottonright` instead of `bottomright`, it will take user a while to figure out what is wrong.

Maybe we should check for this 4 valid kw? I know this doesn't solve it for all templates, but in this case it may be small enough that it is worth it.

Reply:
Do you mean as a static typing check or as an assert?

Reply:
We should avoid `assert` use  [outside of tests](https://docs.astral.sh/ruff/rules/assert/).

I'm old school but here is what I would do:

1. make position a kw argument, not an arg;
2. Sanitize it with a `position.lower()` to validate names copied from leaflet JS docs;
3. Check if the sanitized positions is in a valid list with those 4 names and raise a ValueError if not.
4. 
What do you think?

Reply:
Makes sense.

Reply:
I see we have been inconsistent in banning `assert` :-)